react/src/components/Table/Table.tsx

```diff
@@ -71,34 +72,103 @@ const resolveSex = (sexPhenotype: string) => {
     } else return 'Unknown';
 };
 
-/* flatten data and compute values as needed (note that column display formatting function should not alter values for ease of export) */
-const prepareData = (queryResult: VariantQueryDataResult[]): ResultTableColumns[] => {
-    return queryResult.flatMap(d => {
-        if (d.variant.callsets.length) {
-            //one row per individual per callset
-            return d.variant.callsets
-                .filter(cs => cs.individualId === d.individual.individualId)
-                .map(cs =>
-                    addAdditionalFieldsAndFormatNulls({
-                        ...cs.info,
-                        ...flattenBaseResults(d),
-                    })
-                );
+/* 1, Sort queryResult in ascending order according to variant's chromosome, ref, alt, start,end. 
+2, flatten data and compute values as needed 
+(note that column display formatting function should not alter values for ease of export) */
+const prepareData = (queryResult: VariantQueryDataResult[]): [ResultTableColumns[], number[]] => {
+    const sortedQueryResult = [...queryResult].sort(function (a, b) {
+        if (a.variant.referenceName < b.variant.referenceName) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref < b.variant.ref
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt < b.variant.alt
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt === b.variant.alt &&
+            a.variant.start < b.variant.start
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt === b.variant.alt &&
+            a.variant.start === b.variant.start &&
+            a.variant.end < b.variant.end
```

Comment:
This is not equivalent to the original if-else block. You could verify that with your suggestion, the sorting order is not what we would expect.

Reply:
Ahah - you're right. I wonder if there's another way we can condense it.

Reply:
As I've mentioned in this PR, I can always use formal logic to reduce the original if-else block, but it will not really be readable. Without extensive comments, future developers would have a hard time understanding what the code is doing.

Reply:
Have you looked at this? https://dev.to/markbdsouza/js-sort-an-array-of-objects-on-multiple-columns-keys-2bj1

Reply:
Just to make sure, do we want to sort `ref` or `alt` first?

Reply:
Either is fine!

Reply:
> Have you looked at this? https://dev.to/markbdsouza/js-sort-an-array-of-objects-on-multiple-columns-keys-2bj1

Yes, this is a better way to condense the experession. Thanks very much!